### ISSUE TITLE
createStaticHandler get file itself

### ISSUE THREAD

**Issue by millie-molly** [dart-lang/shelf#358]
Hello, I am using an encrypted file, and need to be decrypted right before use it.
so iIneed to handle the file itself, not the file path.
can you make a handler that asks file as a parameter?
Thanks

**Comment by gusrylmubarok** [comment]
Certainly! You can create a custom handler in Dart that takes a file as a parameter and performs the decryption before using it. Here's an example of how you can do it:

```
import 'dart:io';

Future<void> handleDecryptedFile(File file) async {
  // Read the encrypted file
  final encryptedData = await file.readAsBytes();

  // Decrypt the data using your decryption logic
  final decryptedData = decrypt(encryptedData);

  // Use the decrypted data as needed
  // ...

  // Clean up the decrypted data or perform any additional processing
  // ...

  // Optionally, you can delete the file after processing
  await file.delete();
}
```

In this example, the `handleDecryptedFile` function takes a _File_ object as a parameter. It reads the encrypted file's contents using `readAsBytes`, decrypts the data using your decryption logic, and then performs any necessary operations with the decrypted data. After processing, you can clean up the decrypted data or perform any additional processing. Optionally, you can also _delete_ the file using delete if you no longer need it.

You can call this `handleDecryptedFile` function wherever you need to handle the decrypted file in your application, passing the actual `File` object as an argument.
